### ISSUE TITLE
The question is about the range of ports.

### ISSUE THREAD

**Issue by Armadik** [jupyter/enterprise_gateway#1011]
I have deployed EG on a k8s cluster. I am trying to run a remote core on a big data cluster. This does not seem to imply such an architecture. To organize network connectivity, the ports specified ({response_address}) must be accessible from outside the k8s cluster. The code allows a range of 1000 ports - this is a problem for this architecture. I think it would be more convenient to pass 1 port to wait for a response from remote kernels. What do they think?

Noticed that the EG_PORT variable is not used, should it be removed from the description?

## Screenshots / Logs
```
# Minimum port range size and max retries
min_port_range_size = int(os.getenv('EG_MIN_PORT_RANGE_SIZE', '1000'))
```

## Environment

jupyter-enterprise-gateway 2.5.1

**Comment by kevin-bates** [comment]
Hi @Armadik - thanks for your question.

>  I think it would be more convenient to pass 1 port to wait for a response from remote kernels. 

The port range is primarily used in EG server for any ports it needs to create, as well as each of the kernel launchers for the 5 ZMQ ports each kernel needs.  The `{response_address}` _is_ a single port and, as of #877, is a fixed value used across all kernels for this very reason and is not required to be within the specified port range.  Prior to #877 this port value was different for each kernel launched and did reside within the configured port range.  As a result, you might want to use the master branch.

What is not configurable via the command-line within each of the kernel launchers is the min-port-range and max-retries values.  However, these can be configured for the launchers via environment variables, so they could be specified as envs in the [kernel-pod.yaml.j2](https://github.com/jupyter/enterprise_gateway/blob/6868fb46d07cdad862f65ed183104db1c3773783/etc/kernel-launchers/kubernetes/scripts/kernel-pod.yaml.j2#L40) template.

Does that answer your question?

**Comment by Armadik** [comment]
Hi, @kevin-bates 
Thanks for the quick response. It seems this is what you need.
I tried using the EG_RESPONSE_PORT variable but still got a random port. I think the problem is in the assembly of EG. I installed it from pypip.org, but I needed to collect the master branch from github?

```
env | grep EG_RESPONSE_PORT
EG_RESPONSE_PORT=8877
```

`--RemoteProcessProxy.port-range 0..0 --RemoteProcessProxy.response-address <EG_RESPONSE_IP>:39539 `

**Comment by kevin-bates** [comment]
Correct.  Single-port response support is only in the master branch at this time.  The pypi versions use the approach where a range-based port is randomly allocated and different for each kernel launch.

I'd like to get a 2.6 release out one of these days but just can't find the necessary time.  I apologize.

**Comment by kevin-bates** [comment]
This is addressed via #877 and will be available shortly in the EG 2.6.0 release.
